test/Microsoft.ML.Predictor.Tests/TestGamPublicInterfaces.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Trainers.FastTree;
+using Xunit;
+
+namespace Microsoft.ML.RunTests
+{
+    public class TestGamPublicInterfaces
+    {
+        public TestGamPublicInterfaces()
+        {
+        }
+
+        [Fact]
+        [TestCategory("FastTree")]
+        public void TestGamDirectInstatiation()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            double intercept = 1;
+
+            var binUpperBounds = new double[2][];
+            binUpperBounds[0] = new double[] { 1, 2, 3 };
+            binUpperBounds[1] = new double[] { 4, 5, 6 };
+
+            var binEffects = new double[2][];
+            binEffects[0] = new double[] { 0, 1, 2 };
+            binEffects[1] = new double[] { 2, 1, 0 };
+
+            var gam = new RegressionGamModelParameters(mlContext, binUpperBounds, binEffects, intercept);
```

Comment:
>RegressionGamModelParameters [](start = 26, length = 28)

Add at least two invalid cases:
1. When the count of upper and lower bounds are different
2. When uppper bound is < lower bound.

Reply:
Great cases!